src/main/kotlin/no/item/xp/plugin/GenerateTypeScriptTask.kt

```diff
@@ -1,27 +1,76 @@
 package no.item.xp.plugin
 
 import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
 import javax.inject.Inject
+import javax.xml.parsers.DocumentBuilderFactory
+import no.item.xp.plugin.models.GeneratedField
+import no.item.xp.plugin.models.XmlFile
+import no.item.xp.plugin.models.XmlType
+import no.item.xp.plugin.parser.Parser
+import no.item.xp.plugin.util.GenerateInterfaceName
+import no.item.xp.plugin.writer.TypeScriptWriter
 import org.gradle.api.DefaultTask
 import org.gradle.api.tasks.TaskAction
+import org.w3c.dom.Document
 
 open class GenerateTypeScriptTask @Inject constructor(private val extension: GenerateTypeScriptExtension) : DefaultTask() {
 
     @TaskAction
     fun generateTypeScript() {
       val files = findXmlFiles()
-      println("Found XML files: ")
-      println("--------------------")
-      files.forEach { println(it) }
-      // 2.send dem til en parser
-      // 3.send resultatet til en writer
-      // 4.exit
-    }
-
+      files.forEach { parseFile(it) }
+  }
   private fun findXmlFiles(): Sequence<String> {
     return File(extension.fileDir)
       .walk()
-      .filter { it.isFile && it.extension == "xml" }
+      .filter { it.isFile }
       .map { it.absolutePath }
   }
+
+  private fun parseFile(it: String) {
```

Comment:
`it` is a sort of keyword in Kotlin, and we should probably not use that as a variable name.
```suggestion
  private fun parseFile(filePath: String) {
```